### PR TITLE
Add 'time-out' message when a command timed out

### PR DESCRIPTION
**Pull request by veger**
This PR clarifies that a time-out occurred, solving confusion by the end user. See #630

### CODE REVIEW

**Comment by zmb3** [comment]
I think the executor should return stderr exactly as the process spit out, and we should detect the timeout downstream and display an appropriate message.

Are you up for making these changes?

**Comment by veger** [comment]
Note that only the implements command checks (downstream) if the failing reason is a time-out. Other commands (check 'usage' only) does not (seem to) check this.

Should we update them all?

**Comment by zmb3** [comment]
gorename and tester also check for timeouts.  You could also add a check to usage.

File: lib/implements/implements.js [hunk 1]
```diff
@@ -55,8 +55,11 @@ export default class Implements {
         }
         return this.goconfig.executor.exec(cmd, args, options).then((r) => {
           if (r.error || r.exitcode !== 0 || (r.stderr && r.stderr.trim() !== '')) {
-            if (this.view) {
-              this.view.update('guru failed' + os.EOL + os.EOL + r.stderr.trim())
+              if (r.exitcode === 124) {
+                this.view.update('guru failed: time-out')
```
Comment (zmb3) [inline lib/implements/implements.js]: How about:

```js
guru failed: operation timed out after `${operation.timeout} ms`
```

**Comment by veger** [comment]
Updated 'implements' time-out with nicer message (and fixed position of return statement, found by linter :smile: ) and added similar implementation to 'usage'

**Comment by zmb3** [comment]
Thanks!

**Comment by tpjg** [comment]
Can this be added for more commands? E.g. when go-plus is doing it's lintTest ("go test -c -o {tempdir} .") it may also timeout, especially if CGO is involved.

**Pull request merged**